src/pages/workspace/accounting/qbo/export/QuickbooksCompanyCardExpenseAccountPage.tsx

```diff
@@ -66,21 +71,36 @@ function QuickbooksCompanyCardExpenseAccountPage({policy}: WithPolicyConnections
                                 title={translate('workspace.qbo.defaultVendor')}
                                 wrapperStyle={[styles.ph5, styles.mb3, styles.mt1]}
                                 isActive={Boolean(autoCreateVendor)}
-                                onToggle={(isOn) => Connections.updatePolicyConnectionConfig(policyID, CONST.POLICY.CONNECTIONS.NAME.QBO, CONST.QUICK_BOOKS_CONFIG.AUTO_CREATE_VENDOR, isOn)}
+                                onCloseError={() => Policy.clearQBOErrorField(policyID, CONST.QUICK_BOOKS_CONFIG.AUTO_CREATE_VENDOR)}
+                                onToggle={(isOn) => {
+                                    if (!isOn) {
```

Comment:
Could this be a little bit of logic on the 4th param instead? Like this:

```
Connections.updatePolicyConnectionConfig(policyID, CONST.POLICY.CONNECTIONS.NAME.QBO, CONST.QUICK_BOOKS_CONFIG.EXPORT_COMPANY_CARD_ACCOUNT, isOn && vendors?.[0]?.name);
```

Reply:
we do not expect false as a value for clean up - it should be undefined. but anyway i will refactor it a bit